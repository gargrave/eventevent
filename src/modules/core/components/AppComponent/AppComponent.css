.appWrapper {
  max-width: 960px;
  margin: auto;
  padding: 20px;
}