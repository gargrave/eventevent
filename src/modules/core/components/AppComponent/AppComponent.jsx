// @flow
import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { func, shape } from 'prop-types';

import Routes from '../../Routes';
import styles from './AppComponent.css';

type Props = {
  actions: Object,
};

class AppComponent extends Component<Props> {
  static propTypes = {
    actions: shape({
      setInitialized: func.isRequired,
    }).isRequired,
  }

  componentDidMount() {
    this.props.actions.setInitialized();
  }

  render() {
    return (
      <BrowserRouter>
        <div className={styles.app}>
          <main>
            <Routes />
          </main>
        </div>
      </BrowserRouter>
    );
  }
}

export default AppComponent;
