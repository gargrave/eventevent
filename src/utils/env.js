const ENV = {
  DEV: 'development',
  PROD: 'production',
};
const DEV_ENV_URL = 'localhost:3000';

let currentEnv;
(function setCurrentEnv() {
  if (!currentEnv) {
    const loc = window.location.href;
    if (loc.match(DEV_ENV_URL)) {
      currentEnv = ENV.DEV;
    } else {
      currentEnv = ENV.PROD;
    }
  }
})();

export const isDev = () => currentEnv === ENV.DEV;
export const isProd = () => currentEnv === ENV.PROD;
