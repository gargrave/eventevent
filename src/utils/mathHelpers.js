export const clamp = (val, min, max) => {
  if (val < min) {
    return min;
  }
  if (val > max) {
    return max;
  }
  return val;
};
