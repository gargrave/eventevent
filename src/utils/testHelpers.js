import React from 'react';
import { mount, shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = { ...defaultProps };
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }

  mountComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return mount(<this.component {...props} />);
  }
}
