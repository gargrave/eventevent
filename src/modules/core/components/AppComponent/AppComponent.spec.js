import { ComponentBuilder } from '../../../../utils/testHelpers';

import AppComponent from './AppComponent';

const defaultProps = {
  actions: {
    setInitialized: jest.fn(),
  },
};

const builder = new ComponentBuilder(
  AppComponent,
  defaultProps,
);

describe('AppComponent', () => {
  let component;

  test('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = builder.shallowGetComponent();
  });
});
