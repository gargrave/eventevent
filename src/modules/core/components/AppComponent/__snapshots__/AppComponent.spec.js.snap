// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppComponent
    actions={
        Object {
            "setInitialized": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className={undefined}
>
        <main>
                <Routes />
        </main>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <Routes />
</main>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Routes />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className={undefined}
>
          <main>
                    <Routes />
          </main>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <main>
            <Routes />
</main>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Routes />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
